index.ts

```diff
@@ -46,37 +56,60 @@ async function main () {
       let answer = await rl.question('Your command:')
       if (answer.startsWith("--insert")) {
         let body = answer.replace("--insert ", "")
-        ditto.store.collection("tasks").upsert({
+        const newTask = {
           body,
           isDeleted: false,
           isCompleted: false
-        })
+        }
+        await ditto.store.execute(`
+          INSERT INTO tasks
+          DOCUMENTS (:newTask)`,
+          { newTask })
       }
       if (answer.startsWith("--toggle")) {
         let id = answer.replace("--toggle ", "")
-        ditto.store.collection("tasks")
-        .findByID(id).update((doc) => {
-          let isCompleted = doc.value.isCompleted
-          doc.at("isCompleted").set(!isCompleted)
-        })
+        queryResult = await ditto.store.execute(`
+          SELECT * FROM tasks
+          WHERE _id = :id`,
+          { id }
+        )
+        let newValue = !queryResult.items
+        .map((item) => {
+          return item.value.isCompleted
+        })[0]
+        await ditto.store.execute(`
+          UPDATE tasks
+          SET isCompleted = :newValue
+          WHERE _id = :id`,
+          { id, newValue }
+        )
       }
       if (answer.startsWith("--list")) {
-        console.log(tasks.map((task) => task.value))
+        queryResult = await ditto.store.execute(`
```

Comment:
You don't need to call `execute` here again because the global variable `tasks` should already be updated from your observer callback. We would try to avoid `execute` here since it's blocking